CsharpExtras/Extensions/ArrayExtension2D.cs

```diff
@@ -1,11 +1,33 @@
 ﻿using System;
 using System.Collections.Generic;
 using System.Linq;
+using System.Runtime.CompilerServices;
+using System.Security.Cryptography.X509Certificates;
 
 namespace CsharpExtras.Extensions
 {
     public static class ArrayExtension2D
     {
+        public static int LastUsedRow(this string[,] array)
+        {
+            return array.LastUsedRow(s => !string.IsNullOrWhiteSpace(s));
+        }
+        public static int LastUsedColumn(this string[,] array)
+        {
+            return array.LastUsedColumn(s => !string.IsNullOrWhiteSpace(s));
+        }
+        public static int LastUsedColumn<TVal>(this TVal[,] array, Predicate<TVal> isUsed)

```

Comment:
I would add some documentation on these new methods. I'm assuming they return the index of the last column where all the values match the function, is that right? What do you return if none match?